src/expr/impl/src/scalar/encrypt.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Debug;
+
+use openssl::error::ErrorStack;
+use openssl::symm::{Cipher, Crypter, Mode as CipherMode};
+use risingwave_expr::{function, CryptographyError, CryptographyStage, ExprError, Result};
+
+#[derive(Debug, Clone)]
+enum Algorithm {
+    Blowfish,
+    Aes,
+}
+
+#[derive(Debug, Clone)]
+enum Mode {
+    Cbc,
+    Ecb,
+}
+#[derive(Debug, Clone)]
+enum Padding {
+    Pkcs,
+    None,
+}
+
+#[derive(Debug, Clone)]
+pub struct CipherConfig {
+    algorithm: Algorithm,
+    mode: Mode,
+    padding: Padding,
+    crypt_key: Vec<u8>,
+}
+
+impl CipherConfig {
+    fn parse_cipher_config(key: &[u8], input: &str) -> Result<CipherConfig> {
+        if key.len() != 16 && key.len() != 24 && key.len() != 32 {
+            return Err(ExprError::InvalidParam {
+                name: "key",
+                reason: format!("invalid key length: {}, expect 16, 24 or 32", key.len()).into(),
+            });
+        }
+
+        let parts: Vec<&str> = input.split(&['-', '/'][..]).collect();
```

Comment:
It seems that we can avoid collection and use iterator directly.
```suggestion
        let mut parts = input.split(&['-', '/'][..]);
```